cg/meta/qc_metrics/collect_qc_metrics.py

```diff
@@ -0,0 +1,85 @@
+"""Module for the collect qc metrics meta api."""
+
+import logging
+from typing import Iterator
+
+from housekeeper.store.models import File, Tag
+
+from cg.clients.arnold.api import ArnoldAPIClient
+from cg.clients.janus.api import JanusAPIClient
+from cg.clients.janus.dto.create_qc_metrics_request import (
+    CreateQCMetricsRequest,
+    FilePathAndTag,
+)
+from cg.clients.janus.exceptions import JanusClientError, JanusServerError
+from cg.exc import HousekeeperFileMissingError
+from cg.meta.meta import MetaAPI
+from cg.models.cg_config import CGConfig
+from cg.store.models import Case, Sample
+
+LOG = logging.getLogger(__name__)
+
+
+class CollectQCMetricsAPI(MetaAPI):
+    """Collect qc metrics API."""
+
+    def __init__(self, config: CGConfig):
+        super().__init__(config)
+        self.janus_api: JanusAPIClient = config.janus_api
+        self.arnold_api: ArnoldAPIClient = config.arnold_api
+
+    def get_qc_metrics_file_paths_for_case(self, case_id: str) -> list[File]:
+        """Get the multiqc metrics files for a case."""
+        files: list[File] = self.housekeeper_api.get_files(
+            bundle=case_id, tags=["qc-metrics", "janus"]
+        ).all()
+        if not files:
+            raise HousekeeperFileMissingError(f"No qc metrics files found for {case_id}.")
+        return files
+
+    @staticmethod
+    def get_tag_to_include(tags: list[Tag]) -> str:
+        """Get the multiqc module file tag."""
+        tags_to_exclude: list[str] = ["qc-metrics", "janus", "multiqc"]
+        for tag in tags:
+            if tag.name not in tags_to_exclude:
+                return tag.name
+
+    def get_file_paths_and_tags(self, files: list[File]) -> list[FilePathAndTag]:
+        """Get the file path and tag for the multiqc files."""
+        file_paths_and_tags: list[FilePathAndTag] = []
+        for file_path_and_tags in files:
+            file_path: str = file_path_and_tags.full_path
+            tag: str = self.get_tag_to_include(file_path_and_tags.tags)
+            file_paths_and_tags.append(FilePathAndTag(file_path=file_path, tag=tag))
+        return file_paths_and_tags
+
+    def get_prep_category(self, case_id: str):
+        """Get the prep category for a sample."""
+        samples: list[Sample] = self.status_db.get_samples_by_case_id(case_id=case_id)
```

Comment:
```suggestion
        samples: list[Sample] = self.status_db.get_samples_by_case_id(case_id)
```
Feel free to disregard